tests/test_events/handler/addTab.test.js

```diff
@@ -0,0 +1,163 @@
+    })
+
+    // Should skip if not opened/closed event
+    test("Should skip if not opened/closed event", () => {
+        // Repo is connected & user account is also valid
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        // Pass isTabEvent=false, which means changeEvent.changed.length > 0
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)
+    })
+
+    // Skip if invalid repo id
+    test("Should skip if repoId doesn't exist", () => {
+        const invalid_repo_path = (repoPath).slice(0,5);
+        const configUtil_2 = new Config(invalid_repo_path, configPath);
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(invalid_repo_path);
+        handler.handleTabChangeEvent(true);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)
+    })
+
+    test("Tabs data in positive case", () => {
+        const createdAt = new Date();
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const config_data = readYML(configPath);
+        const tab1_fileId = config_data.repos[repoPath].branches[DEFAULT_BRANCH]["file_1.js"]
+        const repo_id = config_data.repos[repoPath].id;
+        const mockTabs = [
+                {
+                    tabs: [
+                        {
+                            input: {
+                                    uri: {
+                                        path: newFilePath1,
+                                }
+                            }
+                        },
+                        {
+                            input: {
+                                    uri: {
+                                        path: newFilePath2,
+                                }
+                            }
+                        },
+                    ]
+                }
+            ]
+        Object.defineProperty(vscode.window.tabGroups, 'all', {
+            get: jest.fn(() => mockTabs),
+          });
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent()
+        let tabFiles = fs.readdirSync(tabsRepo)
+        // Assert file should be created
+        expect(tabFiles).toHaveLength(1);
+        const tabFilePath = path.join(tabsRepo, tabFiles[0]);
+        const tabData = readYML(tabFilePath);
+        // Assert source == 'vscode'
+        expect(tabData.source).toEqual(VSCODE);
+        // Assert created_at value of tab file and testing value to be in range of 1 second
+        expect((new Date(tabData.created_at)).getTime() - (createdAt).getTime()).toBeLessThanOrEqual(1000);
+        // Assert repo_id
+        expect(tabData.repo_id).toEqual(repo_id);
+        // Assert tabs
+        expect(tabData.tabs[0].file_id).toBe(tab1_fileId);
+        expect(typeof tabData.tabs[0].file_id).toBe('number');
```

Comment:
I think matching with exact FileID is enough, no need to check the type separately